### PR TITLE
Fix race#452

### PR DESCRIPTION
**Pull request by b00ris**
Remove data race warnings.
If you know more elegant solution, please let me know.

Important changes:
- Add mutex guard to signal/notificationHandler

Solves #452

### CODE REVIEW

**Review by JekaMas — APPROVED** [review]

File: geth/signal/signals.go [hunk 1]
```diff
@@ -71,6 +79,8 @@ func Send(signal Envelope) {
 //export NotifyNode
 //nolint: golint
 func NotifyNode(jsonEvent *C.char) {
+	notificationHandlerMutex.RLock()
+	defer notificationHandlerMutex.RUnlock()
```
Comment (divan) [inline geth/signal/signals.go:83]: I'd suggest putting `RUnlock` at the end of the function without using `defer` for consistency with other mutex usage introduced in this PR.

File: geth/signal/signals.go [hunk 2]
```diff
@@ -47,14 +48,21 @@ type NodeNotificationHandler func(jsonEvent string)
 
 var notificationHandler NodeNotificationHandler = TriggerDefaultNodeNotificationHandler
 
+// notificationHandlerMutex needs to remove data races on SetDefaultNodeNotificationHandler,ResetDefaultNodeNotificationHandler
```
Comment (divan) [inline geth/signal/signals.go]: I think a simpler comment like ".. guards `notificationHandler` for concurrent calls...` would be enough — no need to update methods names mentioned in a current comment, for example.

**Review by divan — APPROVED** [review]
Minor remarks. LGTM.

File: geth/signal/signals.go (hunk 1 above)
Comment (b00ris) [reply geth/signal/signals.go:83]: It's my first commit implementation.
My motivation is if notificationHandler panics notificationHandlerMutex will be unlocked.

Comment (divan) [reply geth/signal/signals.go:83]: Ah, that actually makes perfect sense.

**Review by tiabc — APPROVED** [review]
Thanks, let's merge this.

**Comment by tiabc** [comment]
Ah, wait, CI is bad. I've restarted it, let's wait until it's done.

**Comment by adambabik** [comment]
@b00ris this looks like a good candidate to be merged as it has a required number of approvals. Please resolve conflicts.

**Comment by b00ris** [comment]
@adambabik Fixed.

**Comment by adambabik** [comment]
Thanks @b00ris.

**Pull request merged**